tests/device_tests/cardano/test_sign_tx.py

```diff
@@ -157,3 +161,18 @@ def _to_device_actions(client, input_flow):
                 client.debug.press_yes()
             else:
                 raise ValueError("Invalid input action")
+
+
+def _transform_binary_values(structure):
+    """Turn all the binary values in the structure into their hex string representation.
+
+    This makes them comparable with the JSON fixtures.
+    Only recurses into nested dicts, lists. Turns nested tuples into lists."""
+    if (is_dict := isinstance(structure, dict)) or isinstance(structure, list):
+        for key, value in structure.items() if is_dict else enumerate(structure):
+            if isinstance(value, bytes):
+                structure[key] = value.hex()
+            else:
+                if isinstance(value, tuple):
+                    structure[key] = list(value)
+                _transform_binary_values(structure[key])
```

Comment:
This selectively recursive transformation is not the cleanest thing I've written, but I don't see structure independent (future proof) testing possible without something like this.